### ISSUE TITLE
Error when copying LVM container from one host to another

### ISSUE THREAD

**Issue by zulcss** [lxc/lxd#1212]
When copying a LVM snapshot from one to host to another, an error will occur:

nova@juju-zulcss-machine-11:~$ lxc copy chuck juju-zulcss-machine-10:chuck
error: The container already exists on disk, container: 'chuck', path: '/var/lib/lxd/containers/chuck'

This will happen when the container does not exist on the new host.

**Comment by stgraber** [comment]
@mikemccracken can you look at this ASAP?

**Comment by mikemccracken** [comment]
FWIW, I can reproduce this by spinning up two LXDs with current master on wily, and setting up LVM storage on both. 

With two systems, source and dest:
```bash
scripts/lxd-setup-storage -s 10G # on both
lxd-images import busybox --alias busybox
lxc config set core.https_address "[::]:8443" 
lxc config set core.trust_password foo
# on source:
lxc remote add dest dest.maas

lxc launch busybox zul # on source
lxc stop zul
lxc copy --debug zul dest:zul
```

**Comment by mikemccracken** [comment]
Here's the debug output from the client on `source` 
```
 lxc copy --debug zul pb:zul
DBUG[10-13|21:57:11] Fingering the daemon 
DBUG[10-13|21:57:11] Raw response: {"type":"sync","status":"Success","status_code":200,"metadata":{"api_compat":1,"auth":"trusted","config":{"core.https_address":"[::]:8443","core.trust_password":true,"images.remote_cache_expiry":"10","storage.lvm_thinpool_name":"LXDPool","storage.lvm_vg_name":"LXDStorage"},"environment":{"addresses":["172.16.0.65:8443","10.0.3.1:8443"],"architectures":[2,1],"driver":"lxc","driver_version":"1.1.4","kernel":"Linux","kernel_architecture":"x86_64","kernel_version":"4.2.0-12-generic","server":"lxd","server_pid":5932,"server_version":"0.19","storage":"lvm","storage_version":"2.02.122(2) (2015-06-20) / 1.02.99 (2015-06-20) / 4.33.0"}}}
 
DBUG[10-13|21:57:11] Pong received 
DBUG[10-13|21:57:11] Raw response: {"type":"sync","status":"Success","status_code":200,"metadata":{"architecture":0,"config":{"volatile.base_image":"940a9632d7b51ac53a7558521197df93de9dab09a35bab3a01cf243aab5fbadd","volatile.eth0.hwaddr":"00:16:3e:5b:05:d0","volatile.last_state.idmap":"[{\"Isuid\":true,\"Isgid\":false,\"Hostid\":1000000,\"Nsid\":0,\"Maprange\":65536},{\"Isuid\":false,\"Isgid\":true,\"Hostid\":1000000,\"Nsid\":0,\"Maprange\":65536}]"},"devices":{},"ephemeral":false,"expanded_config":{"volatile.base_image":"940a9632d7b51ac53a7558521197df93de9dab09a35bab3a01cf243aab5fbadd","volatile.eth0.hwaddr":"00:16:3e:5b:05:d0","volatile.last_state.idmap":"[{\"Isuid\":true,\"Isgid\":false,\"Hostid\":1000000,\"Nsid\":0,\"Maprange\":65536},{\"Isuid\":false,\"Isgid\":true,\"Hostid\":1000000,\"Nsid\":0,\"Maprange\":65536}]"},"expanded_devices":{"eth0":{"hwaddr":"00:16:3e:5b:05:d0","nictype":"bridged","parent":"lxcbr0","type":"nic"}},"name":"zul","profiles":["default"],"status":{"status":"Stopped","status_code":102,"init":0,"ips":null}}}
 
DBUG[10-13|21:57:11] Fingering the daemon 
DBUG[10-13|21:57:11] Raw response: {"type":"sync","status":"Success","status_code":200,"metadata":{"api_compat":1,"auth":"trusted","config":{"core.https_address":"[::]:8443","core.trust_password":true,"images.remote_cache_expiry":"10","storage.lvm_thinpool_name":"LXDPool","storage.lvm_vg_name":"LXDStorage"},"environment":{"addresses":["172.16.0.64:8443","10.0.3.1:8443"],"architectures":[2,1],"driver":"lxc","driver_version":"1.1.4","kernel":"Linux","kernel_architecture":"x86_64","kernel_version":"4.2.0-12-generic","server":"lxd","server_pid":15067,"server_version":"0.19","storage":"lvm","storage_version":"2.02.122(2) (2015-06-20) / 1.02.99 (2015-06-20) / 4.33.0"}}}
 
DBUG[10-13|21:57:11] Pong received 
DBUG[10-13|21:57:11] Raw response: {"type":"sync","status":"Success","status_code":200,"metadata":["/1.0/profiles/default","/1.0/profiles/migratable"]}
 
DBUG[10-13|21:57:11] Posting {"migration":true}
 to http://unix.socket/1.0/containers/zul 
DBUG[10-13|21:57:11] Raw response: {"type":"async","status":"OK","status_code":100,"operation":"/1.0/operations/24de8e56-d967-4a55-a515-e964ee2d7770","resources":null,"metadata":{"control":"1a40dfd0964fda2c689d0867f5f59fb4a0fc502c3362be736e95a8d7128a97bd","fs":"673ddc9375a73c64d39919c09b6eb56062a8eeeb8fb24296e7b9e215e5c31122"}}
 
DBUG[10-13|21:57:11] Raw response: {"type":"sync","status":"Success","status_code":200,"metadata":{"api_compat":1,"auth":"trusted","config":{"core.https_address":"[::]:8443","core.trust_password":true,"images.remote_cache_expiry":"10","storage.lvm_thinpool_name":"LXDPool","storage.lvm_vg_name":"LXDStorage"},"environment":{"addresses":["172.16.0.65:8443","10.0.3.1:8443"],"architectures":[2,1],"driver":"lxc","driver_version":"1.1.4","kernel":"Linux","kernel_architecture":"x86_64","kernel_version":"4.2.0-12-generic","server":"lxd","server_pid":5932,"server_version":"0.19","storage":"lvm","storage_version":"2.02.122(2) (2015-06-20) / 1.02.99 (2015-06-20) / 4.33.0"}}}
 
DBUG[10-13|21:57:11] Posting {"config":{},"ephemeral":false,"name":"zul","profiles":["default"],"source":{"base-image":"940a9632d7b51ac53a7558521197df93de9dab09a35bab3a01cf243aab5fbadd","mode":"pull","operation":"wss://172.16.0.65:8443/1.0/operations/24de8e56-d967-4a55-a515-e964ee2d7770/websocket","secrets":{"control":"1a40dfd0964fda2c689d0867f5f59fb4a0fc502c3362be736e95a8d7128a97bd","fs":"673ddc9375a73c64d39919c09b6eb56062a8eeeb8fb24296e7b9e215e5c31122"},"type":"migration"}}
 to https://petty-blade.maas:8443/1.0/containers 
DBUG[10-13|21:57:11] Raw response: {"type":"async","status":"OK","status_code":100,"operation":"/1.0/operations/8d7e01a6-a754-4aec-8e63-ea3f5d56b36f","resources":{"containers":["/1.0/containers/zul"]},"metadata":null}
 
DBUG[10-13|21:57:11] 1.0/operations/8d7e01a6-a754-4aec-8e63-ea3f5d56b36f/wait 
DBUG[10-13|22:00:11] Raw response: {"type":"sync","status":"Success","status_code":200,"metadata":{"created_at":"2015-10-13T21:57:11.767410731Z","updated_at":"2015-10-13T22:00:11.283805018Z","status":"Failure","status_code":400,"resources":{"containers":["/1.0/containers/zul"]},"metadata":"Error transferring container data: exit status 23","may_cancel":false}}
 
DBUG[10-13|22:00:11] Posting {"config":{},"ephemeral":false,"name":"zul","profiles":["default"],"source":{"base-image":"940a9632d7b51ac53a7558521197df93de9dab09a35bab3a01cf243aab5fbadd","mode":"pull","operation":"wss://10.0.3.1:8443/1.0/operations/24de8e56-d967-4a55-a515-e964ee2d7770/websocket","secrets":{"control":"1a40dfd0964fda2c689d0867f5f59fb4a0fc502c3362be736e95a8d7128a97bd","fs":"673ddc9375a73c64d39919c09b6eb56062a8eeeb8fb24296e7b9e215e5c31122"},"type":"migration"}}
 to https://petty-blade.maas:8443/1.0/containers 
DBUG[10-13|22:00:11] Raw response: {"type":"async","status":"OK","status_code":100,"operation":"/1.0/operations/74b2a3fc-1a69-4cb4-acf5-a7e4f210775f","resources":{"containers":["/1.0/containers/zul"]},"metadata":null}
 
DBUG[10-13|22:00:11] 1.0/operations/74b2a3fc-1a69-4cb4-acf5-a7e4f210775f/wait 
DBUG[10-13|22:02:04] Raw response: {"type":"sync","status":"Success","status_code":200,"metadata":{"created_at":"2015-10-13T22:00:11.289641469Z","updated_at":"2015-10-13T22:02:04.427820217Z","status":"Failure","status_code":400,"resources":{"containers":["/1.0/containers/zul"]},"metadata":"Error transferring container data: websocket: bad handshake","may_cancel":false}}
 
error: Error transferring container data: websocket: bad handshake
```

**Comment by mikemccracken** [comment]
Here's the debug output from the server:
```
.... (I got lazy pasting the first part, the important part is later)
                                "addresses": [                                                                                                                                            [127/2307]
                                        "172.16.0.64:8443",
                                        "10.0.3.1:8443"
                                ],
                                "architectures": [
                                        2,
                                        1
                                ],
                                "driver": "lxc",
                                "driver_version": "1.1.4",
                                "kernel": "Linux",
                                "kernel_architecture": "x86_64",
                                "kernel_version": "4.2.0-12-generic",
                                "server": "lxd",
                                "server_pid": 15067,
                                "server_version": "0.19",
                                "storage": "lvm",
                                "storage_version": "2.02.122(2) (2015-06-20) / 1.02.99 (2015-06-20) / 4.33.0"
                        }
                }
        } 
DBUG[10-13|21:57:11] Found cert                               k=0
INFO[10-13|21:57:11] handling                                 method=GET url=/1.0/profiles ip=172.16.0.65:41674
DBUG[10-13|21:57:11] 
        {
                "type": "sync",
                "status": "Success",
                "status_code": 200,
                "metadata": [
                        "/1.0/profiles/default",
                        "/1.0/profiles/migratable"
                ]
        } 
DBUG[10-13|21:57:11] Found cert                               k=0
INFO[10-13|21:57:11] handling                                 method=POST url=/1.0/containers ip=172.16.0.65:41674
DBUG[10-13|21:57:11] 
        {
                "config": {},
                "ephemeral": false,
                "name": "zul",
                "profiles": [
                        "default"
                ],
                "source": {
                        "base-image": "940a9632d7b51ac53a7558521197df93de9dab09a35bab3a01cf243aab5fbadd",
                        "mode": "pull",
                        "operation": "wss://172.16.0.65:8443/1.0/operations/24de8e56-d967-4a55-a515-e964ee2d7770/websocket",
                        "secrets": {
                                "control": "1a40dfd0964fda2c689d0867f5f59fb4a0fc502c3362be736e95a8d7128a97bd",
                                "fs": "673ddc9375a73c64d39919c09b6eb56062a8eeeb8fb24296e7b9e215e5c31122"
                        },
                        "type": "migration"
                }
        } 
DBUG[10-13|21:57:11] Responding to container create 
DBUG[10-13|21:57:11] 
        {
                "type": "async",
                "status": "OK",
                "status_code": 100,
                "operation": "/1.0/operations/8d7e01a6-a754-4aec-8e63-ea3f5d56b36f",
                "resources": {
                        "containers": [
                                "/1.0/containers/zul"
                        ]
                },
                "metadata": null
        } 
DBUG[10-13|21:57:11] ******************************************************************************** create as empty 
INFO[10-13|21:57:11] Container create                         container=zul isSnapshot=false
DBUG[10-13|21:57:11] Found cert                               k=0
INFO[10-13|21:57:11] handling                                 method=GET url=/1.0/operations/8d7e01a6-a754-4aec-8e63-ea3f5d56b36f/wait ip=172.16.0.65:41674
DBUG[10-13|21:57:11] Container created in the DB              container=zul id=6
DBUG[10-13|21:57:11] ContainerCreate                          driver=storage/lvm isPrivileged=false name=zul
DBUG[10-13|22:00:10] about to start lvm storage for container cname=zul
DBUG[10-13|22:00:10] ContainerStart                           driver=storage/lvm container=zul
DBUG[10-13|22:00:10] ContainerStop                            driver=storage/lvm container=zul
DBUG[10-13|22:00:10] Got error getting next reader read tcp 172.16.0.64:58858->172.16.0.65:8443: use of closed network connection, &{%!s(*os.File=&{0xc820159bf0}) {{%!s(int32=0) %!s(uint32=0)} %!s
(uint32=1)} <nil>} 
DBUG[10-13|22:00:11] containerLXD.Delete                      c.name=zul type=0
DBUG[10-13|22:00:11] containerDeleteSnapshots                 container=zul
DBUG[10-13|22:00:11] ContainerDelete                          driver=storage/lvm container=zul
DBUG[10-13|22:00:11] Operation %!s(func() shared.OperationResult=0x555290) finished: { Error transferring container data: exit status 23} 
DBUG[10-13|22:00:11] 
        {
                "type": "sync",
                "status": "Success",
                "status_code": 200,
                "metadata": {
                        "created_at": "2015-10-13T21:57:11.767410731Z",
                        "updated_at": "2015-10-13T22:00:11.283805018Z",
                        "status": "Failure",
                        "status_code": 400,
                        "resources": {
                                "containers": [
                                        "/1.0/containers/zul"
                                ]
                        },
                        "metadata": "Error transferring container data: exit status 23",
                        "may_cancel": false
                }
        } 
DBUG[10-13|22:00:11] Found cert                               k=0
INFO[10-13|22:00:11] handling                                 method=POST url=/1.0/containers ip=172.16.0.65:41674
DBUG[10-13|22:00:11] 
        {
                "config": {},
                "ephemeral": false,
                "name": "zul",
                "profiles": [
                        "default"
                ],
                "source": {
                        "base-image": "940a9632d7b51ac53a7558521197df93de9dab09a35bab3a01cf243aab5fbadd",
                        "mode": "pull",
                        "operation": "wss://10.0.3.1:8443/1.0/operations/24de8e56-d967-4a55-a515-e964ee2d7770/websocket",
                        "secrets": {
                                "control": "1a40dfd0964fda2c689d0867f5f59fb4a0fc502c3362be736e95a8d7128a97bd",
                                "fs": "673ddc9375a73c64d39919c09b6eb56062a8eeeb8fb24296e7b9e215e5c31122"
                        },
                        "type": "migration"
                }
        } 
DBUG[10-13|22:00:11] Responding to container create 
DBUG[10-13|22:00:11] 
        {
                "type": "async",
                "status": "OK",
                "status_code": 100,
                "operation": "/1.0/operations/74b2a3fc-1a69-4cb4-acf5-a7e4f210775f",
                "resources": {
                        "containers": [
                                "/1.0/containers/zul"
                        ]
                },
                "metadata": null
        } 
DBUG[10-13|22:00:11] ******************************************************************************** create as empty 
INFO[10-13|22:00:11] Container create                         container=zul isSnapshot=false
DBUG[10-13|22:00:11] Found cert                               k=0
INFO[10-13|22:00:11] handling                                 method=GET url=/1.0/operations/74b2a3fc-1a69-4cb4-acf5-a7e4f210775f/wait ip=172.16.0.65:41674
DBUG[10-13|22:00:11] Container created in the DB              container=zul id=7
DBUG[10-13|22:00:11] ContainerCreate                          driver=storage/lvm isPrivileged=false name=zul
DBUG[10-13|22:02:04] about to start lvm storage for container cname=zul
DBUG[10-13|22:02:04] ContainerStart                           driver=storage/lvm container=zul
DBUG[10-13|22:02:04] Client cert != key                       k=0
INFO[10-13|22:02:04] allowing untrusted GET                   url="/1.0/operations/24de8e56-d967-4a55-a515-e964ee2d7770/websocket?secret=1a40dfd0964fda2c689d0867f5f59fb4a0fc502c3362be736e95a8d7128a97bd" ip=10.0.3.1:46254
DBUG[10-13|22:02:04] 
        {
                "error": "not found",
                "error_code": 404,
                "type": "error"
        } 
DBUG[10-13|22:02:04] Raw response: {"error":"not found","error_code":404,"type":"error"}

 
DBUG[10-13|22:02:04] ContainerStop                            driver=storage/lvm container=zul
DBUG[10-13|22:02:04] containerLXD.Delete                      type=0 c.name=zul
DBUG[10-13|22:02:04] containerDeleteSnapshots                 container=zul
DBUG[10-13|22:02:04] ContainerDelete                          driver=storage/lvm container=zul
DBUG[10-13|22:02:04] Operation %!s(func() shared.OperationResult=0x555290) finished: { Error transferring container data: websocket: bad handshake} 
DBUG[10-13|22:02:04] 
        {
                "type": "sync",
                "status": "Success",
                "status_code": 200,
                "metadata": {
                        "created_at": "2015-10-13T22:00:11.289641469Z",
                        "updated_at": "2015-10-13T22:02:04.427820217Z",
                        "status": "Failure",
                        "status_code": 400,
                        "resources": {
                                "containers": [
                                        "/1.0/containers/zul"
                                ]
                        },
                        "metadata": "Error transferring container data: websocket: bad handshake",
                        "may_cancel": false
                }
        } 
```

**Comment by mikemccracken** [comment]
Note, the LVM-unmount related bug was fixed by #1213, but the actual transfer bug remains and we are still looking at it.
